frontend/src/widgets/importancecomponent.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  Card,
+  CardBody,
+  Text
+} from '@patternfly/react-core';
+
+import {
+  Table,
+  Thead,
+  Th,
+  Tbody,
+  Tr,
+  Td
+} from '@patternfly/react-table';
+
+import { Link } from 'react-router-dom';
+
+import { HttpClient } from '../services/http';
+import { Settings } from '../settings';
+import { WidgetHeader } from '../components/widget-components';
+
+export class ImportanceComponentWidget extends React.Component {
+  static propTypes = {
+    title: PropTypes.string,
+    params: PropTypes.object,
+    onDeleteClick: PropTypes.func,
+    onEditClick: PropTypes.func
+  }
+
+  constructor(props) {
+    super(props);
+    this.title = props.title || 'Importance Component Widget';
+    this.params = props.params || {};
+    this.state = {
+      data: {
+        table_data: []
+      },
+      isLoading: true,
+    };
+  }
+
+  getData = () => {
+    this.setState({isLoading: true})
+    HttpClient.get([Settings.serverUrl, 'widget', 'importance-component'], this.params)
+      .then(response => {
+        response = HttpClient.handleResponse(response, 'response');
+        if (!response.ok) {
+          throw Error(response.statusText);
+        }
+        return response.json();
+      })
+      .then(data => this.setState({data: data, isLoading: false}))
+      .catch(error => {
+        this.setState({dataError: true});
+        console.log(error);
+      });
+  }
+
+  componentDidMount() {
+    this.getData();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.params !== this.props.params) {
+      this.params = this.props.params;
+      this.getData();
+    }
+  }
+
+  render() {
+    return (
+      <Card>
+        <WidgetHeader title={this.title} getDataFunc={this.getData} onEditClick={this.props.onEditClick} onDeleteClick={this.props.onDeleteClick}/>
+        {(!this.state.dataError && this.state.isLoading) &&
+        <CardBody>
+          <Text component="h2">Loading ...</Text>
+        </CardBody>
+        }
+        {(!this.state.dataError && !this.state.isLoading) &&
+        <CardBody>
+          {this.state.data.table_data.map((tdat) => (
+            <>
+              <Text key={tdat.component} component="h2">{tdat.component}</Text>
+              <Table aria-label="tttable" variant="compact">
```

Comment:
What's up with the label here? Something more descriptive would be good.

Reply:
Oops, added that while drafting this out and forgot to update it. Should be fixed now.